File: fixtures/scale/objc/large/Families/Gronsfeld/GronsfeldDescriptor.m
#import "GronsfeldDescriptor.h"
#import "GronsfeldCipher.h"
#import "GronsfeldVectors.h"

@implementation GronsfeldDescriptor

- (NSString *)family {
    return @"gronsfeld";
}

- (NSString *)suite {
    return @"polyalphabetic";
}

- (id<CLCipher>)cipher {
    return [[GronsfeldCipher alloc] init];
}

- (NSArray *)vectors {
    return [GronsfeldVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Gronsfeld/GronsfeldDescriptor.php
<?php

namespace CipherLab\Families\Gronsfeld;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together GronsfeldCipher and
// GronsfeldVectors under the "polyalphabetic" suite.
class GronsfeldDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'gronsfeld';
    }

    public function suite(): string
    {
        return 'polyalphabetic';
    }

    public function cipher(): Cipher
    {
        return new GronsfeldCipher();
    }

    public function vectors(): array
    {
        return GronsfeldVectors::all();
    }
}
